packages/nimbus/src/components/alert/alert.dev.mdx

```diff
@@ -0,0 +1,269 @@
+
+**Title only:**
+
+```jsx-live-dev
+const App = () => (
+  <Alert.Root colorPalette="positive">
+    <Alert.Title>Success: Item created.</Alert.Title>
+  </Alert.Root>
+)
+```
+
+**Description only:**
+
+```jsx-live-dev
+const App = () => (
+  <Alert.Root colorPalette="info">
+    <Alert.Description>
+      System maintenance scheduled for tonight at 02:00 UTC.
+    </Alert.Description>
+  </Alert.Root>
+)
+```
+
+### Adding actions
+
+Use `Alert.Actions` to group action buttons. These will be positioned correctly within the alert layout.
+
+```jsx-live-dev
+const App = () => (
+  <Alert.Root colorPalette="warning">
+    <Alert.Title>Storage Limit Reached</Alert.Title>
+    <Alert.Description>
+      You have used 95% of your available storage.
+    </Alert.Description>
+    <Alert.Actions>
+      <Stack direction="row" gap="200">
+        <Button variant="solid">Upgrade Plan</Button>
+        <Button variant="ghost">Dismiss</Button>
+      </Stack>
+    </Alert.Actions>
+  </Alert.Root>
+)
+```
+
+### Dismissible alerts
+
+Add the `Alert.DismissButton` to provide a close action. You must handle the dismissal logic (e.g., removing the alert from the DOM) yourself.
+
+```jsx-live-dev
+const App = () => {
+  const [isVisible, setIsVisible] = useState(true);
+
+  if (!isVisible) {
+    return (
+      <Button onPress={() => setIsVisible(true)}>
+        Reset Example
+      </Button>
+    );
+  }
+
+  return (
+    <Alert.Root colorPalette="info" variant="outlined">
+      <Alert.Title>Dismissible Alert</Alert.Title>
+      <Alert.Description>
+        Click the close button to dismiss this message.
+      </Alert.Description>
+      <Alert.DismissButton onPress={() => setIsVisible(false)} />
+    </Alert.Root>
+  );
+}
+```
+
+## Component requirements
+
+### Accessibility
+
+The `Alert` component has built-in accessibility features:
+
+- **Role**: The root element automatically has `role="alert"`. This informs screen readers that the content is important and time-sensitive.
+- **Icons**: Decorative icons are automatically hidden from screen readers to avoid redundancy.
+- **Dismiss Button**: The `Alert.DismissButton` comes with a default accessible label ("Dismiss") which can be localized or overridden via `aria-label`.
+
+If your use case requires tracking and analytics for this component, it is good practice to add a **persistent**, **unique** id to the component:
+
+```tsx
+const PERSISTENT_ID = "example-alert";
+
+export const Example = () => (
+  <Alert.Root id={PERSISTENT_ID}>
+    <Alert.Title>Notification</Alert.Title>
+  </Alert.Root>
+);
+```
+
+### Composition constraints
+
+- All sub-components (`Title`, `Description`, `Actions`, `DismissButton`) **must** be direct children of `Alert.Root` to function correctly. They communicate via context to register themselves for layout.
```

Comment:
Similar to the `Card`, I think we should at least consider using css to achieve the expected layout instead of context.  This would mean that no matter what the sub-components are wrapped in, the grid would put them into the correct layout.

See [this article](https://marijkeluttekes.dev/blog/articles/2025/01/12/conditional-css-grid-template-areas/) for how we could potentially achieve this